packages/keystone/src/admin-ui/components/Pagination.tsx

```diff
@@ -43,6 +44,21 @@ export function Pagination({ currentPage, total, pageSize, list }: PaginationPro
   const limit = Math.ceil(total / pageSize);
   const pages = [];
 
+  useEffect(() => {
+    // Check if the current page is larger than
+    // the maximal page given the total and associated page size value.
+    // (This could happen due to a deletion event, in which case we want to reroute the user to a previous page).
+    if (currentPage > Math.ceil(total / pageSize)) {
+      push({
+        pathname,
+        query: {
+          ...query,
+          page: currentPage - 1,
```

Comment:
Without the below suggestion, if I add a filter that lowers the number of pages, this will decrease the page number one by one instead of jumping to the last available one.

```suggestion
          page: Math.ceil(total / pageSize),
```

Reply:
ah nice, good catch